File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.cpp
class Solution {
public:
    int xorAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        int res = 0;
        for(int i = 0; i < queries.size(); i++)
        {
            int idx = queries[i][0];
            while(idx <= queries[i][1])
            {
                nums[idx] = ((long long)nums[idx] * queries[i][3]) % 1000000007;
                idx += queries[i][2];
            }
        }
        for(int i = 0; i < nums.size(); i++)
        {
            res ^= nums[i];
        }
        return res;
    }
};

File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.py
class Solution:
    def xorAfterQueries(self, nums: List[int], queries: List[List[int]]) -> int:
        res = 0
        for l, r, k, v in queries:
            idx = l
            while idx <= r:
                nums[idx] = (nums[idx] * v) % 1000000007
                idx += k
        for num in nums:
            res ^= num
        return res